src/views/home/components/transaction-form/transaction-form.view.tsx

```diff
@@ -14,72 +13,118 @@ import Button from "src/views/shared/button/button.view";
+interface FormChains {
   from: Chain;
   to: Chain;
-  token: Token;
-  amount?: BigNumber;
 }
 
 const TransactionForm: FC<TransactionFormProps> = ({ onSubmit, transaction, account }) => {
-  const classes = useTransactionFormtStyles();
+  const classes = useTransactionFormStyles();
   const env = useEnvContext();
+  const { openSnackbar } = useUIContext();
+  const { estimateBridgeGasPrice } = useBridgeContext();
   const [list, setList] = useState<List>();
-  const [error, setError] = useState<string>();
-  const [balanceFrom, setBalanceFrom] = useState(BigNumber.from(0));
-  const [balanceTo, setBalanceTo] = useState(BigNumber.from(0));
-  const [formData, setFormData] = useState<FormData | undefined>(transaction);
+  const [balanceFrom, setBalanceFrom] = useState<BigNumber>();
+  const [balanceTo, setBalanceTo] = useState<BigNumber>();
+  const [inputError, setInputError] = useState<string>();
+
+  const [chains, setChains] = useState<FormChains>();
+  const [token, setToken] = useState<Token>();
+  const [amount, setAmount] = useState<BigNumber>();
+  const [estimatedFee, setEstimatedFee] = useState<AsyncTask<BigNumber, string>>({
+    status: "pending",
+  });
 
   const onChainFromButtonClick = (from: Chain) => {
-    if (env && formData) {
+    if (env && chains) {
       const to = env.chains.find((chain) => chain.key !== from.key);
+
       if (to) {
-        setFormData({ ...formData, from, to, amount: undefined });
+        setChains({ from, to });
         setList(undefined);
       }
     }
   };
 
   const onInputChange = ({ amount, error }: { amount?: BigNumber; error?: string }) => {
-    if (formData && amount) {
-      setFormData({ ...formData, amount });
-      setError(error);
-    }
+    setAmount(amount);
+    setInputError(error);
   };
 
   const onFormSubmit = (e: React.FormEvent) => {
     e.preventDefault();
-    if (formData && formData.amount) {
+    if (chains && token && amount && estimatedFee.status === "successful") {
       onSubmit({
-        ...formData,
-        amount: formData.amount,
+        token: token,
+        from: chains.from,
+        to: chains.to,
+        amount: amount,
+        estimatedFee: estimatedFee.data,
       });
     }
   };
 
   useEffect(() => {
-    void formData?.from.provider.getBalance(account).then(setBalanceFrom);
-    void formData?.to.provider.getBalance(account).then(setBalanceTo);
-  }, [formData?.from, formData?.to, account]);
+    if (transaction !== undefined) {
+      setChains({ from: transaction.from, to: transaction.to });
+      setToken(transaction.token);
+      setAmount(transaction.amount);
+    } else if (env !== undefined) {
+      setChains({ from: env.chains[0], to: env.chains[1] });
+      setToken(env.tokens.ETH);
+    }
+  }, [env, transaction]);
 
   useEffect(() => {
-    if (env && !transaction) {
-      setFormData({
-        from: env.chains[0],
-        to: env.chains[1],
-        token: env.tokens.ETH,
-      });
+    setAmount(undefined);
```

Comment:
The previous effect is in charge of restoring the data from the `TransactionData` present in the context, including the Chains. Whenever the chains are updated, this effect is executed, resetting the amount back to `undefined` and thus breaking the behavior of retrieving the amount entered by the user when the Confirm Transfer screen is closed.